src/aggregate.d

```diff
@@ -270,13 +276,15 @@ extern (C++) abstract class AggregateDeclaration : ScopeDsymbol
             return 0;
         }
 
-        fields.setDim(0);
-
         for (size_t i = 0; i < members.dim; i++)
         {
             auto s = (*members)[i];
             if (s.apply(&func, cast(void*)this))
+            {
+                if (sizeok != SIZEOKnone)
+                    return true;
```

Comment:
Yeah, looks like that is from 2.067.  That should be no longer needed, as `determineFields` and `setFieldOffset` should really work together, rather than stomp on each others toes.

If you look at the (roughly sketched) call trace: https://github.com/dlang/dmd/pull/6531#discussion_r100687964 - notice that `fields` gets reset to 0 three times, and the third time it is fully populated.